test/src/tests.js

```diff
@@ -139,4 +174,577 @@ describe('Bing Ads Event Forwarder', function() {
             done();
         });
     });
+
+    describe('Consent', function() {
+        var consentMap = [
+            {
+                jsmap: null,
+                map: 'marketing_consent',
+                maptype: 'ConsentPurposes',
+                value: 'ad_storage',
+            },
+        ];
+
+        beforeEach(function() {
+            mParticle.forwarders = [];
+        });
+
+        afterEach(function() {
+            window.uetq = [];
+        });
+
+        it('should consolidate consent information', function(done) {
+            mParticle.forwarder.init(
+                {
+                    tagId: 'tagId',
+                },
+                reportService.cb,
+                false // Disable testMode so we can test init
+            );
+
+            var obj = {
+                EventDataType: MessageType.PageEvent,
+                EventName: 'Test Page Event',
+                CustomFlags: {
+                    'Bing.EventValue': 10,
+                },
+                ConsentState: {
+                    getGDPRConsentState: function() {
+                        return {
+                            some_consent: {
+                                Consented: true,
+                                Timestamp: 1557935884509,
+                                ConsentDocument: 'fake_consent_document',
+                                Location: 'This is fake',
+                                HardwareId: '123456',
+                            },
+                        };
+                    },
+                },
+            };
+
+            var expectedConsentPayload = [
+                'consent',
+                'update',
+                { ad_storage: 'granted' },
+            ];
+
+            var expectedEventPayload = {
+                ea: 'pageLoad',
+                ec: 'This is my name!',
+                el: 'Test Page Event',
+                ev: 10,
+            };
+
+            mParticle.forwarder.process(obj);
+
+            // UETQ queues up events as array elements and then parses them internally.
+            // The first 3 elements of this array will be the consent payload
+            // The 4th element will be the event payload
+            window.uetq.length.should.eql(4);
+            window.uetq[0].should.equal('consent');
+            window.uetq[1].should.equal('default');
+            window.uetq[2].should.eql(expectedConsentPayload[2]);
+            window.uetq[3].should.eql(expectedEventPayload);
+
+            done();
+        });
+
+        it('should construct a Default Consent State Payload from Mappings on init', function(done) {
```

Comment:
Add to the title that there is no `defaultAdStorageConsentWeb` setting. That's what this test is actually testing.